metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java

```diff
@@ -130,43 +124,46 @@ private Model getMetadata(URI uri) throws ExtractionException, IOException, URIS
     private Model normalizeStatements(RepositoryResult<Statement> statements) throws IOException, InterruptedException {
         Model model = new TreeModel();
         for (Statement statement : statements) {
-            if (isDoi(statement)) {
-                Optional<Statement> doi = toBiboDoi(statement);
-                doi.ifPresent(model::add);
-            } else if (isSindiceDcOrDcTerms(statement)) {
-                model.add(toDctermsNamespace(statement));
-            } else if (isHighwireDate(statement)) {
-                model.add(toDctermsDate(statement));
-            } else if (isHighWireLanguage(statement)) {
-                model.add(toDctermsLanguage(statement));
-            } else if (isHighWireAuthor(statement)) {
-                model.add(toDctermsCreator(statement));
-            } else {
-                model.add(statement);
-            }
+            extractKnownProperties(model, statement);
         }
         return model;
     }
 
-    private Statement toDctermsCreator(Statement statement) {
-        return valueFactory.createStatement(statement.getSubject(),
-                DcTerms.CREATOR.getIri(valueFactory), statement.getObject());
+    private void extractKnownProperties(Model model, Statement statement) throws IOException, InterruptedException {
+        OntologyProperty ontologyProperty = getMappedOntologyProperty(statement);
+        Value value = extractValue(ontologyProperty, statement.getObject());
+        OntologyProperty mappedProperty = mapToSpecificProperty(ontologyProperty, value);
+        if (nonNull(ontologyProperty)) {
```

Comment:
We're not interested in properties we can't do anything with in the SPARQL query, since this is the case, we might also consider using a more programmatic approach iterating over the triples.